Library/Homebrew/env_config.rb

```diff
@@ -132,9 +132,9 @@ module EnvConfig
         boolean:     true,
       },
       HOMEBREW_CURLRC:                           {
-        description: "If set, do not pass `--disable` when invoking `curl`(1), which disables the " \
-                     "use of `curlrc`.",
-        boolean:     true,
+        description: "If set to a path, pass it with `--config` when invoking `curl`(1). " \
+                     "If set but _not_ a valid path, do not pass `--disable`, which disables the " \
+                     "use of `.curlrc`.",
```

Comment:
This is great. Wondering if this is the best approach or if we want to do something like "if set to `1` or `true`"? @Homebrew/brew thoughts?

Reply:
That's a very fair question! The current handling of `boolean: true` is `#present?`, but it might be nice to enforce something stricter in this particular case.

Reply:
> This is great. Wondering if this is the best approach or if we want to do something like "if set to `1` or `true`"? @Homebrew/brew thoughts?

Starting with `/` is probably sufficient to invoke the path-based logic.

There's potentially also an argument that we could just let `CURL_HOME` passthrough if `HOMEBREW_CURLRC` is set rather than the double-purpose, but it maybe doesn't matter much.

Reply:
> Starting with `/` is probably sufficient to invoke the path-based logic, given it is unlikely someone does that in order to use it as a boolean and we won't ever support relative paths.

Agreed, that works for me!